transport/http_util.go

```diff
@@ -159,6 +161,8 @@ func newHPACKDecoder() *hpackDecoder {
 			}
 		case ":path":
 			d.state.method = f.Value
+		case "content-type":
```

Comment:
The design for this feature was such that the client explicitly specifies to use a non-standard codec. It'd be nice just from the symmetry perspective to have the server be similarly explicit. I don't think it's a particularly necessary goal that a server can automatically switch between wire formats, but maybe you're thinking about that?

If we're going for some arbitrary string-matching scheme here, the codec package (or wherever that code lives) will probably need a registration mechanism. We can't bake in knowledge about every possible codec.

The use of the codec's `String()` result in `Content-Type` needs explicit documentation in the `Codec` interface definition.